hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -3230,7 +3235,9 @@ void transferReplicaForPipelineRecovery(final ExtendedBlock b,
     final BlockConstructionStage stage;
 
     //get replica information
-    try(AutoCloseableLock lock = data.acquireDatasetReadLock()) {
+
+    try(AutoCloseableLock lock = dataSetLockManager.writeLock(
```

Comment:
this was a read lock before. Any idea why it is made a write lock now?

Reply:
this should be read lock.I write wrong here.